File: 250627/각 자리 숫자의 제곱/square-of-each-digit.java
import java.util.Scanner;
public class Main {
    public static void main(String[] args) {
        Scanner sc = new Scanner(System.in);
        int n = sc.nextInt();
        // Please write your code here.


        System.out.println(pow(n));
    }

    public static int pow(int n){
        if(n < 10) {
            return (int)Math.pow(n,2);
        }

        return pow(n/10) + (int)Math.pow(n%10,2);
    }
}

File: 250627/각 자리 숫자의 제곱/square-of-each-digit.py
import math;

N = int(input())

# Please write your code here.

# 각 자리 수의 제곱을 반환하라.
# 크게보면 an = (a%10) ^ 2 + (a % 10) ^ 2 하면 됨

def pow_num(N) :
    if N < 10 :
        return pow(N,2);


    return pow_num(N//10) + pow(N % 10, 2);


print(pow_num(N));